graph_altimeter/scan/__init__.py

```diff
@@ -172,17 +174,25 @@ def _fix_orphan_edges(graph_dict, scan_id):
     vertices = graph_dict["vertices"]
     edges = graph_dict["edges"]
 
-    existing_vertices = {v["~id"] for v in vertices}
+    existing_vertices = {v["~id"]: v for v in vertices}
     non_existing_vertices = {}
     for e in edges:
         from_vid = e["~from"]
         to_vid = e["~to"]
+        # Skip error vertices.
         if ((from_vid not in existing_vertices) and
            (from_vid not in non_existing_vertices)):
-            non_existing_vertices[from_vid] = _create_vertex(from_vid, scan_id)
+            if existing_vertices[to_vid]["~label"] == "error":
+                account_id = ""
+            else:
+                account_id = existing_vertices[to_vid]["account_id"]
+            non_existing_vertices[from_vid] = \
+                _create_vertex(from_vid, scan_id, account_id)
         if ((to_vid not in existing_vertices) and
            (to_vid not in non_existing_vertices)):
-            non_existing_vertices[to_vid] = _create_vertex(to_vid, scan_id)
+            account_id = existing_vertices[from_vid]["account_id"]
```

Comment:
How do you know that `existing_vertices[from_vid]` exists?

Reply:
Yes!!